Production/scripts/ShuffleMergeLaw/analysis/tasks.py

```diff
@@ -78,3 +78,62 @@ def run(self):
     retcode = proc.returncode
     if proc.returncode != 0:
       raise Exception('job {} return code is {}'.format(self.branch, retcode))
+
+class HaddFiles(Task, HTCondorWorkflow, law.LocalWorkflow):
```

Comment:
- I think HaddFiles should be part of a separate workflow (e.g. CrabPostProc). We need to do it only once after the production is done and not each time we re-run ShffleMerge.
- If some files are corrupted, hadd still exits with code 0. To catch such cases, you need to search in the log files word "error" or "warning" (case insensitive) and if it is found, consider the job as failed.

Reply:
Do you think adding -k flag (Skip corrupt or non-existent files) would be an option? and then we can parse output files after that to create the list of corrupted files?

Reply:
Yes, having `-k` flag is a good idea. Like this, in the log file we'll have all corrupted files (if there are more than one). However, in my opinion, if there is at least one corrupted file, the condor job should be considered as failed and after the problem is solved resubmitted again. I.e. if HaddFiles is done just after the crab production, the corresponding crab job can be resubmitted. If not, one can decide to remove the corrupted files.